ring/ring_operations.go

```diff
@@ -396,6 +414,16 @@ func (r *Ring) MulScalarBigintLvl(level int, p1 *Poly, scalar *big.Int, p2 *Poly
 	}
 }
 
+// EvalPolMontgomeryScalarNTT evaluate the polynomial pol at pk and writes the result in p3
+func (r *Ring) EvalPolMontgomeryScalarNTT(pol []*Poly, pk uint64, p3 *Poly) {
```

Comment:
Rename to EvalPolyScalarMontgomery if that's the scalar that is supposed to be in Montgomery